crates/flox/src/commands/channel.rs

```diff
@@ -144,50 +146,136 @@ fn construct_search_params(search_term: &str, flox: &Flox) -> Result<SearchParam
+        .map(|d| {
+            if d.join {
+                vec![d.input.as_str(), d.package.as_str()]
+                    .join(SEPARATOR)
+                    .len()
+            } else {
+                d.package.len()
+            }
+        })
+        .max()
+        .unwrap(); // SAFETY: could panic if `deduped_display_items` is empty, but we know it's not
+
+    // Finally print something
     let mut writer = BufWriter::new(std::io::stdout());
-    for (attr, desc) in summarized_results.iter() {
-        writeln!(&mut writer, "{attr:<attr_col_width$}  {desc}")?;
+    let default_desc = String::from("<no description provided>");
+    for d in deduped_display_items.into_iter() {
+        let package = if d.join {
+            vec![d.input, d.package].join(SEPARATOR)
+        } else {
+            d.package
+        };
+        let desc: String = d.description.unwrap_or(default_desc.clone());
+        writeln!(&mut writer, "{package:<column_width$}  {desc}")?;
     }
     Ok(())
 }
 
-fn render_search_results_json(search_results: &SearchResults) -> Result<()> {
+fn render_search_results_json(search_results: SearchResults) -> Result<()> {
     let json = serde_json::to_string(&search_results.results)?;
     println!("{}", json);
     Ok(())
 }
 
+/// Deduplicate and disambiguate display items.
+///
+/// This gets complicated because we have to satisfy a few constraints:
+/// - The order of results from `pkgdb` is important (best matches come first),
+///   so that order must be preserved.
+/// - Versions shouldn't appear in the output, so multiple package versions from a single
+///   input should be deduplicated.
+/// - Packages that appear in more than one input need to be disambiguated by prepending
+///   the name of the input and a separator.
+fn dedup_and_disambiguate_display_items(mut display_items: Vec<DisplayItem>) -> Vec<DisplayItem> {
+    let mut package_to_inputs: HashMap<String, HashSet<String>> = HashMap::new();
+    for d in display_items.iter() {
+        // Build a collection of packages and which inputs they are seen in
+        package_to_inputs
+            .entry(d.package.clone())
+            .and_modify(|inputs| {
+                inputs.insert(d.input.clone());
+            })
+            .or_insert_with(|| {
+                let mut inputs = HashSet::new();
+                inputs.insert(d.input.clone());
+                inputs
+            });
+    }
+
+    // For any package that comes from more than one input, mark it as needing to be joined
+    for d in display_items.iter_mut() {
+        if let Some(inputs) = package_to_inputs.get(&d.package) {
+            d.join = inputs.len() > 1 || d.join;
+        }
+    }
+
+    let mut deduped_display_items = Vec::new();
+    for d in display_items.into_iter() {
+        if let Some(inputs) = package_to_inputs.get_mut(d.package.as_str()) {
+            // Remove this input so this (input, package) pair is never seen again
```

Comment:
Just some docs suggestions to figure out what's going on in the future (TM)
```suggestion
    // `package_to_inputs` practically represents `{ (package, input) }`.
    // If a DisplayItem is in `package_to_inputs` we have not yet seen it.
    // We add it to `deduped_display_items`
    // and remove (package, input) from `package_to_inputs`
    // so following occurences of (package, input) are skipped.
    let mut deduped_display_items = Vec::new();
    for d in display_items.into_iter() {
        if let Some(inputs) = package_to_inputs.get_mut(&d.package) {
            // Remove this input so this (package, input) pair is never seen again
```

Reply:
Fixed, thanks